subsys/bluetooth/host/smp.c

```diff
@@ -26,13 +26,14 @@
 #include <zephyr/sys/byteorder.h>
 #include <zephyr/sys/util.h>
 
-#include <bt_crypto.h>
 
 #define BT_DBG_ENABLED	IS_ENABLED(CONFIG_BT_DEBUG_SMP)
 #define LOG_MODULE_NAME bt_smp
 #include "common/log.h"
 #include "common/bt_str.h"
 
+#include "crypto/bt_crypto.h"
```

Comment:
bt_crypto.h is one level up relative to folder of this folder.
```suggestion
#include "../crypto/bt_crypto.h"
```
or:
```suggestion
#include <bt_crypto.h>
```

Reply:
This one I disagree with. See all the `"common/log.h"` and `"common/bt_str.h"`. We include `subsys/bluetooth` to the include path for internal files: 
https://github.com/zephyrproject-rtos/zephyr/blob/c9e07fc38515351aebd25d87f3e87f0d3e1dcfc9/subsys/bluetooth/CMakeLists.txt#L5

Reply:
that CMake line actually places the include paths as `INTERFACE`.

Interface means used by everyone linking to `subsys__bluetooth` will inherit this include path, but the library itself **will not use** that include paths.

So if this linked line is intended as you describe then it's wrong, cause this is telling the outside world: `Use this include directory when linking to this library, this is my public header location.`

Reply:
why do you disagree with:
```c
#include "../crypto/bt_crypto.h"
```
that exactly indicates it's a local header, and that this particular header is located one level up, and then down to `crypto/` folder seen from current location.

Reply:
Yep, you are right. I think the right line is here: https://github.com/zephyrproject-rtos/zephyr/blob/c9e07fc38515351aebd25d87f3e87f0d3e1dcfc9/subsys/bluetooth/CMakeLists.txt#L19. Which is anyway wrong (we shouldn't be placing that folder in the global include path). But fixing this will require more work, so let's leave it for a follow-up PR.

Reply:
> why do you disagree with:
> `#include "../crypto/bt_crypto.h"`

Because it is inconsistent with what we are doing elsewhere in the subsystem. See `common/log.h` for example, or any other folder in `common/`

Reply:
> Because it is inconsistent with what we are doing elsewhere in the subsystem. See common/log.h for example, or any other folder in common/

and the reason it works is exactly because the other flaw in the CMake code.
So we have two flawed principles in use which cancels out each other so things end up working.

This is actually not nice, as it just makes it harder to cleanup, generally.
For example, I cannot cleanup the CMake code, because the source code relies on the flawed CMake.

But as cleaning up this would be outside scope of this PR, so staying consistent with other parts of this file is accepted.